src/main/scala/bi/deep/DruidSchema.scala

```diff
@@ -2,24 +2,18 @@ package bi.deep
 
 import org.apache.druid.segment.column.ColumnCapabilitiesImpl
 
-class DruidSchema(
-                   var dimensions: Array[(String, ColumnCapabilitiesImpl)],
-                   var metrics: Array[(String, ColumnCapabilitiesImpl)]) {
+case class DruidSchema(
```

Comment:
It is a bad practice to construct an object and then fix it later. The rule of thumb is that:
If you can create an object in a valid state now, then do it now. No need for further "validate" etc.

In this case we could have private constructor and an apply method in companion object to build the object
in correct state already. In our case, we care about concrete naming convection so enforce them on 
object creation.

You can actually improve it like this:

```scala
case class DruidSchema private (
    dimensions: Array[(String, ColumnCapabilitiesImpl)],
    metrics: Array[(String, ColumnCapabilitiesImpl)]
) 

object DruidSchema {
    def apply(
        dimensions: Array[(String, ColumnCapabilitiesImpl)],
        metrics: Array[(String, ColumnCapabilitiesImpl)]
    ): DruidSchema = new DruidSchema(dimensions.map(fix), metrics.map(fix))


   /** Fixes the column names in the provided schema and removes all invalid characters, such as:
    * `,`, `;`, `{`, `}`, `(`, `)`, `=`, and all whitespace characters.
    * */
    private def fix(pair: (String, ColumnCapabilitiesImpl)): (String, ColumnCapabilitiesImpl) = pair match {
         case (name, capabilities) => name.replaceAll("[\\s,;{}()=]", "") -> capabilities
    }
}
```